src/io/benchmark.zig

```diff
@@ -0,0 +1,240 @@
+const std = @import("std");
+const os = std.os;
+const assert = std.debug.assert;
+
+const Time = @import("time.zig").Time;
+const IO = @import("io.zig").IO;
+
+pub fn main() !void {
+    try struct {
+        io: IO,
+        tx: Pipe,
+        rx: Pipe,
+        timer: *Time,
+        started: u64,
+        current: u64,
+        server: Socket = .{},
+        transferred: u64 = 0,
+
+        // 1 MB: larger than socket buffer so forces io_pending on darwin
+        // Cnofigure this value to smaller amounts to test IO scheduling overhead
+        const buffer_size = 1 * 1024 * 1024;
+
+        // max time for the benchmark to run
+        const run_duration = 1 * std.time.ns_per_s;
+
+        const Context = @This();
+        const Socket = struct {
+            fd: os.socket_t = -1,
+            completion: IO.Completion = undefined,
+        };
+        const Pipe = struct {
+            socket: Socket = .{},
+            buffer: []u8,
+            transferred: usize = 0,
+        };
+
+        fn run() !void {
+            var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+            const allocator = &gpa.allocator;
+            defer assert(gpa.deinit());
+
+            const buf = try allocator.alloc(u8, buffer_size * 2);
+            defer allocator.free(buf);
+            std.mem.set(u8, buf, 0);
+            
+            var timer = Time{};
+            const started = timer.monotonic();
+            var self = Context{
+                .io = try IO.init(32, 0),
+                .timer = &timer,
+                .started = started,
+                .current = started,
+                .tx = .{ .buffer = buf[0*buffer_size..][0..buffer_size] },
```

Comment:
I think if we run `zig fmt` it will spread out the `0*buffer_size`? If not already run, let's quickly run `zig fmt` on this.

Reply:
Yup, it did.